core/src/erasure.rs

```diff
@@ -138,23 +138,25 @@ impl Session {
 
             if n < NUM_DATA {
                 let mut blob = Blob::new(&blocks[n]);
+                blob.meta.size = blob.data_size() as usize;
```

Comment:
@pgarg66 can you prove to yourself these are the right steps with size?

Reply:
It seems ok to me. The following is my analysis:

In old code it's calling `blob.set_size()` with value `size - BLOB_HEADER_SIZE`.
```
    pub fn set_size(&mut self, size: usize) {
        let new_size = size + BLOB_HEADER_SIZE;
        self.meta.size = new_size;
        self.set_data_size(new_size as u64);
    }
```

`set_size` adds back the `BLOB_HEADER_SIZE` and updates the meta. In addition, it calls `set_data_size()`. But that's not needed, as we are reading the size from blob itself. So it should already be correct in the blob header.